internal/controllers/machine/machine_controller.go

```diff
@@ -372,7 +380,86 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Clu
 				return ctrl.Result{}, errors.Wrap(err, "failed to patch Machine")
 			}
 
-			if result, err := r.drainNode(ctx, cluster, m.Status.NodeRef.Name); !result.IsZero() || err != nil {
+			result, err := r.drainNode(ctx, cluster, drainNodeInput{
+				NodeName: m.Status.NodeRef.Name,
+				OnPodDeletionOrEvictionFinished: func(pod *corev1.Pod, usingEviction bool, drainErr error) {
```

Comment:
Let move this those inline handler functions to proper func (not inline) so we can write a small unit test

Reply:
I have added unit tests too, please check.